commit 1fbeb54f7dd6c9d4579a7f9fae0e949e48013302
Author: Pratyush Anand <panand@redhat.com>
Date:   Mon May 29 22:08:24 2017 +0300

    mei: make sysfs modalias format similar as uevent modalias
    
    commit 6f9193ec044a8f72d8b6ae94a5c4ab6e8b0f00ca upstream.
    
    modprobe is not able to resolve sysfs modalias for mei devices.
    
     # cat
    /sys/class/watchdog/watchdog0/device/watchdog/watchdog0/device/modalias
    mei::05b79a6f-4628-4d7f-899d-a91514cb32ab:
     # modprobe --set-version 4.9.6-200.fc25.x86_64 -R
    mei::05b79a6f-4628-4d7f-899d-a91514cb32ab:
    modprobe: FATAL: Module mei::05b79a6f-4628-4d7f-899d-a91514cb32ab: not
    found in directory /lib/modules/4.9.6-200.fc25.x86_64
     # cat /lib/modules/4.9.6-200.fc25.x86_64/modules.alias | grep
    05b79a6f-4628-4d7f-899d-a91514cb32ab
    alias mei:*:05b79a6f-4628-4d7f-899d-a91514cb32ab:*:* mei_wdt
    
    commit b26864cad1c9 ("mei: bus: add client protocol
    version to the device alias"), however sysfs modalias
    is still in formmat mei:S:uuid:*.
    
    This patch equates format of uevent and sysfs modalias so that modprobe
    is able to resolve the aliases.
    
    Fixes: commit b26864cad1c9 ("mei: bus: add client protocol version to the device alias")
    Signed-off-by: Pratyush Anand <panand@redhat.com>
    Signed-off-by: Tomas Winkler <tomas.winkler@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/misc/mei/bus.c b/drivers/misc/mei/bus.c
index dbe676d..0c98ed4 100644
--- a/drivers/misc/mei/bus.c
+++ b/drivers/misc/mei/bus.c
@@ -678,8 +678,10 @@ static ssize_t modalias_show(struct device *dev, struct device_attribute *a,
 {
 	struct mei_cl_device *cldev = to_mei_cl_device(dev);
 	const uuid_le *uuid = mei_me_cl_uuid(cldev->me_cl);
+	u8 version = mei_me_cl_ver(cldev->me_cl);
 
-	return scnprintf(buf, PAGE_SIZE, "mei:%s:%pUl:", cldev->name, uuid);
+	return scnprintf(buf, PAGE_SIZE, "mei:%s:%pUl:%02X:",
+			 cldev->name, uuid, version);
 }
 static DEVICE_ATTR_RO(modalias);
 
